src/skimage/measure/fit.py

```
@@ -904,6 +904,16 @@ def _estimate(self, data, warn_only=True):
         # from this equation [eqn. 28]
         eig_vals, eig_vecs = np.linalg.eig(M)
 
+        # https://github.com/scikit-image/scikit-image/issues/7013
+        if np.any(eig_vals.imag != 0) or np.any(eig_vecs.imag != 0):
+            raise TypeError(
+                "Non-real eigvalues/eigvec are not currently supported. Please "
+                "report the reproducer to "
+                "https://github.com/scikit-image/scikit-image/issues/7013"
+            )
```

Ruff (0.15.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)